src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,567 @@
+        }
+        else {
+            log.warn("Cannot include text submission content in data export because content is null for submission with id: {}", textSubmission.getId());
+        }
+    }
+
+    private void createResultsTxtFile(Submission submission, Path outputDir) throws IOException {
+        StringBuilder resultScoreAndFeedbacks = new StringBuilder();
+        for (var result : submission.getResults()) {
+            if (result != null) {
+                var score = result.getScore();
+                if (score != null) {
+                    resultScoreAndFeedbacks.append("Score of submission: ").append(score).append("%").append(" ")
+                            .append(score * submission.getParticipation().getExercise().getMaxPoints() / 100).append(" Points").append("\n");
+                }
+                if (submission instanceof ProgrammingSubmission && result.getPassedTestCaseCount() != null && result.getTestCaseCount() != null && result.getTestCaseCount() > 0) {
+                    resultScoreAndFeedbacks.append("Passed test cases: ").append(result.getPassedTestCaseCount()).append("/").append(result.getTestCaseCount()).append("\n");
+                }
+                for (var feedback : result.getFeedbacks()) {
+                    resultScoreAndFeedbacks.append("- Feedback: ");
+                    // null if it's manual feedback
+                    if (feedback.getText() != null) {
+                        resultScoreAndFeedbacks.append(feedback.getText()).append("\t");
+                    }
+                    // null if the test case passes
+                    if (feedback.getDetailText() != null) {
+                        resultScoreAndFeedbacks.append(feedback.getDetailText()).append("\t");
+                    }
+                    resultScoreAndFeedbacks.append(feedback.getCredits()).append("\n");
+                }
+                Files.writeString(outputDir.resolve("submission_" + submission.getId() + "_result_" + result.getId() + TXT_FILE_EXTENSION), resultScoreAndFeedbacks);
+            }
+            resultScoreAndFeedbacks = new StringBuilder();
+        }
+
+    }
+
+    private void createNonProgrammingExerciseExport(Exercise exercise, Path courseDir) throws IOException {
+        Path exercisePath = courseDir.resolve(exercise.getSanitizedExerciseTitle());
+        if (!Files.exists(exercisePath)) {
+            Files.createDirectory(exercisePath);
+        }
+        createSubmissionsResultsExport(exercise, exercisePath);
+
+    }
+
+    private void copyFileUploadSubmissionFile(String submissionFilePath, Path outputDir) {
+        try {
+            FileUtils.copyDirectory(new File(submissionFilePath), outputDir.toFile());
+        }
+        catch (IOException ignored) {
+            // ignore if we cannot retrieve the submitted file, it might no longer exist.
```

Comment:
Maybe include something in the export that the user knows that this file is no longer there?